--- ../src-base/minecraft/net/minecraft/world/gen/structure/WoodlandMansion.java
+++ ../src-work/minecraft/net/minecraft/world/gen/structure/WoodlandMansion.java
@@ -1,5 +1,7 @@
 package net.minecraft.world.gen.structure;
 
+import carpet.logging.LoggerRegistry;
+import carpet.logging.logHelpers.RNGMonitor;
 import com.google.common.collect.Lists;
 import java.util.Arrays;
 import java.util.List;
@@ -9,6 +11,7 @@
 import net.minecraft.util.Rotation;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.biome.BiomeProvider;
 import net.minecraft.world.chunk.ChunkPrimer;
@@ -54,6 +57,12 @@
         k = k + (random.nextInt(60) + random.nextInt(60)) / 2;
         l = l + (random.nextInt(60) + random.nextInt(60)) / 2;
 
+        // CM start
+        if (LoggerRegistry.__rngManip && field_75039_c instanceof WorldServer) {
+            ((WorldServer) field_75039_c).rngMonitor.tryUpdateSeeds(RNGMonitor.RNGAppType.raw);
+        }
+        // CM end
+
         if (p_75047_1_ == k && p_75047_2_ == l)
         {
             boolean flag = this.field_75039_c.func_72959_q().func_76940_a(p_75047_1_ * 16 + 8, p_75047_2_ * 16 + 8, 32, field_191072_a);
