--- ../src-base/minecraft/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
@@ -1,14 +1,15 @@
 package net.minecraft.world.chunk.storage;
 
-import com.google.common.collect.Maps;
+import carpet.helpers.lifetime.spawning.LiteralSpawningReason;
+import com.google.common.collect.Maps; //CM unused import
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.File;
 import java.io.IOException;
-import java.util.Collections;
+import java.util.Collections; //CM unused import
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
+import java.util.Set; //CM unused import
 import javax.annotation.Nullable;
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
@@ -34,12 +35,62 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.HashMap;
+import java.util.Iterator;
+import carpet.CarpetSettings;
+
 public class AnvilChunkLoader implements IChunkLoader, IThreadedFileIO
 {
     private static final Logger field_151505_a = LogManager.getLogger();
-    private final Map<ChunkPos, NBTTagCompound> field_75828_a = Maps.<ChunkPos, NBTTagCompound>newConcurrentMap();
-    private final Set<ChunkPos> field_193415_c = Collections.<ChunkPos>newSetFromMap(Maps.newConcurrentMap());
-    private final File field_75825_d;
+    //CM removed old structures
+    //private final Map<ChunkPos, NBTTagCompound> chunksToRemove = Maps.<ChunkPos, NBTTagCompound>newConcurrentMap();
+    //private final Set<ChunkPos> chunksBeingSaved = Collections.<ChunkPos>newSetFromMap(Maps.newConcurrentMap());
+    //CM added new stuctures that will be managed separately
+    private final Map<ChunkPos, NBTTagCompound> field_75828_a = new HashMap();
+    private final Map<ChunkPos, NBTTagCompound> chunksInWrite = new HashMap();
+    // Insert new chunk into pending queue, replacing any older one at the same position
+    synchronized private void queueChunkToRemove(ChunkPos pos, NBTTagCompound data)
+    {
+        field_75828_a.put(pos, data);
+    }
+    // Fetch another chunk to save to disk and atomically move it into
+    // the queue of chunk(s) being written.
+    synchronized private Map.Entry<ChunkPos, NBTTagCompound> fetchChunkToWrite()
+    {
+        if (field_75828_a.isEmpty()) return null;
+        Iterator<Map.Entry<ChunkPos, NBTTagCompound>> iter =
+                field_75828_a.entrySet().iterator();
+        Map.Entry<ChunkPos, NBTTagCompound> entry = iter.next();
+        iter.remove();
+        chunksInWrite.put(entry.getKey(), entry.getValue());
+        return entry;
+    }
+
+    // Once the write for a chunk is completely committed to disk,
+    // this method discards it
+    synchronized private void retireChunkToWrite(ChunkPos pos, NBTTagCompound data)
+    {
+        chunksInWrite.remove(pos);
+    }
+
+    // Check these data structures for a chunk being reloaded
+    synchronized private NBTTagCompound reloadChunkFromRemoveQueues(ChunkPos pos)
+    {
+        NBTTagCompound data = field_75828_a.get(pos);
+        if (data != null) return data;
+        return (CarpetSettings.entityDuplicationFix)?chunksInWrite.get(pos):data;
+    }
+
+    // Check if chunk exists at all in any pending save state
+    //synchronized private boolean chunkExistInRemoveQueues(ChunkPos pos)
+    //{
+    //    return chunksToRemove.containsKey(pos) || chunksInWrite.containsKey(pos);
+    //}
+
+    /* --- end of new code for MC-119971 --- */
+
+
+    public final File field_75825_d; // CM changed to public for ticking areas
     private final DataFixer field_193416_e;
     private boolean field_183014_e;
 
@@ -50,14 +101,15 @@
     }
 
     @Nullable
-    public Chunk func_75815_a(World p_75815_1_, int p_75815_2_, int p_75815_3_) throws IOException
+    public Chunk loadChunk(World worldIn, int x, int z, boolean silent) throws IOException
     {
-        ChunkPos chunkpos = new ChunkPos(p_75815_2_, p_75815_3_);
-        NBTTagCompound nbttagcompound = this.field_75828_a.get(chunkpos);
+        ChunkPos chunkpos = new ChunkPos(x, z);
+        //NBTTagCompound nbttagcompound = this.chunksToRemove.get(chunkpos);
+        NBTTagCompound nbttagcompound = reloadChunkFromRemoveQueues(chunkpos);  // new for MC-119971
 
         if (nbttagcompound == null)
         {
-            DataInputStream datainputstream = RegionFileCache.func_76549_c(this.field_75825_d, p_75815_2_, p_75815_3_);
+            DataInputStream datainputstream = RegionFileCache.func_76549_c(this.field_75825_d, x, z);
 
             if (datainputstream == null)
             {
@@ -67,43 +119,46 @@
             nbttagcompound = this.field_193416_e.func_188257_a(FixTypes.CHUNK, CompressedStreamTools.func_74794_a(datainputstream));
         }
 
-        return this.func_75822_a(p_75815_1_, p_75815_2_, p_75815_3_, nbttagcompound);
+        return this.checkedReadChunkFromNBT(worldIn, x, z, nbttagcompound, silent);
     }
 
     public boolean func_191063_a(int p_191063_1_, int p_191063_2_)
     {
         ChunkPos chunkpos = new ChunkPos(p_191063_1_, p_191063_2_);
-        NBTTagCompound nbttagcompound = this.field_75828_a.get(chunkpos);
+        //CM
+        //NBTTagCompound nbttagcompound = this.chunksToRemove.get(chunkpos);
+        NBTTagCompound nbttagcompound = reloadChunkFromRemoveQueues(chunkpos);
+        //
         return nbttagcompound != null ? true : RegionFileCache.func_191064_f(this.field_75825_d, p_191063_1_, p_191063_2_);
     }
 
     @Nullable
-    protected Chunk func_75822_a(World p_75822_1_, int p_75822_2_, int p_75822_3_, NBTTagCompound p_75822_4_)
+    protected Chunk checkedReadChunkFromNBT(World worldIn, int x, int z, NBTTagCompound compound, boolean silent)
     {
-        if (!p_75822_4_.func_150297_b("Level", 10))
+        if (!compound.func_150297_b("Level", 10))
         {
-            field_151505_a.error("Chunk file at {},{} is missing level data, skipping", Integer.valueOf(p_75822_2_), Integer.valueOf(p_75822_3_));
+            field_151505_a.error("Chunk file at {},{} is missing level data, skipping", Integer.valueOf(x), Integer.valueOf(z));
             return null;
         }
         else
         {
-            NBTTagCompound nbttagcompound = p_75822_4_.func_74775_l("Level");
+            NBTTagCompound nbttagcompound = compound.func_74775_l("Level");
 
             if (!nbttagcompound.func_150297_b("Sections", 9))
             {
-                field_151505_a.error("Chunk file at {},{} is missing block data, skipping", Integer.valueOf(p_75822_2_), Integer.valueOf(p_75822_3_));
+                field_151505_a.error("Chunk file at {},{} is missing block data, skipping", Integer.valueOf(x), Integer.valueOf(z));
                 return null;
             }
             else
             {
-                Chunk chunk = this.func_75823_a(p_75822_1_, nbttagcompound);
+                Chunk chunk = this.readChunkFromNBT(worldIn, nbttagcompound, silent);
 
-                if (!chunk.func_76600_a(p_75822_2_, p_75822_3_))
+                if (!chunk.func_76600_a(x, z))
                 {
-                    field_151505_a.error("Chunk file at {},{} is in the wrong location; relocating. (Expected {}, {}, got {}, {})", Integer.valueOf(p_75822_2_), Integer.valueOf(p_75822_3_), Integer.valueOf(p_75822_2_), Integer.valueOf(p_75822_3_), Integer.valueOf(chunk.field_76635_g), Integer.valueOf(chunk.field_76647_h));
-                    nbttagcompound.func_74768_a("xPos", p_75822_2_);
-                    nbttagcompound.func_74768_a("zPos", p_75822_3_);
-                    chunk = this.func_75823_a(p_75822_1_, nbttagcompound);
+                    field_151505_a.error("Chunk file at {},{} is in the wrong location; relocating. (Expected {}, {}, got {}, {})", Integer.valueOf(x), Integer.valueOf(z), Integer.valueOf(x), Integer.valueOf(z), Integer.valueOf(chunk.field_76635_g), Integer.valueOf(chunk.field_76647_h));
+                    nbttagcompound.func_74768_a("xPos", x);
+                    nbttagcompound.func_74768_a("zPos", z);
+                    chunk = this.readChunkFromNBT(worldIn, nbttagcompound, silent);
                 }
 
                 return chunk;
@@ -132,37 +187,63 @@
 
     protected void func_75824_a(ChunkPos p_75824_1_, NBTTagCompound p_75824_2_)
     {
-        if (!this.field_193415_c.contains(p_75824_1_))
+        //CM extended condition to mimic vanilla better
+        //if (!this.chunksBeingSaved.contains(pos))
+        //{
+        //    this.chunksToRemove.put(pos, compound);
+        //}
+        if ( CarpetSettings.entityDuplicationFix || !chunksInWrite.containsKey(p_75824_1_))
         {
-            this.field_75828_a.put(p_75824_1_, p_75824_2_);
+            queueChunkToRemove(p_75824_1_, p_75824_2_);
         }
 
+
         ThreadedFileIOBase.func_178779_a().func_75735_a(this);
     }
 
     public boolean func_75814_c()
     {
-        if (this.field_75828_a.isEmpty())
+/*        if (this.chunksToRemove.isEmpty())
         {
-            if (this.field_183014_e)
+            if (this.savingExtraData)
             {
-                field_151505_a.info("ThreadedAnvilChunkStorage ({}): All chunks are saved", (Object)this.field_75825_d.getName());
+                LOGGER.info("ThreadedAnvilChunkStorage ({}): All chunks are saved", (Object)this.chunkSaveLocation.getName());
             }
 
             return false;
         }
         else
         {
-            ChunkPos chunkpos = this.field_75828_a.keySet().iterator().next();
-            boolean lvt_3_1_;
+            */
 
-            try
-            {
-                this.field_193415_c.add(chunkpos);
-                NBTTagCompound nbttagcompound = this.field_75828_a.remove(chunkpos);
+            //ChunkPos chunkpos = this.chunksToRemove.keySet().iterator().next();
 
-                if (nbttagcompound != null)
+            Map.Entry<ChunkPos, NBTTagCompound> entry = fetchChunkToWrite();
+            if (entry == null) {
+                // If none left, here's code for some message that will never
+                // be executed since there is no "extra data."
+                if (this.field_183014_e)
                 {
+                    field_151505_a.info("ThreadedAnvilChunkStorage ({}): All chunks are saved", new Object[] {this.field_75825_d.getName()});
+                }
+
+                return false;
+            }
+
+            // New for MC-119971
+            ChunkPos chunkpos = entry.getKey();
+            NBTTagCompound nbttagcompound = entry.getValue();
+
+
+            //boolean lvt_3_1_;
+
+            //try
+            //{
+            //   this.chunksBeingSaved.add(chunkpos);
+            //    NBTTagCompound nbttagcompound = this.chunksToRemove.remove(chunkpos);
+
+            //    if (nbttagcompound != null)
+            //    {
                     try
                     {
                         this.func_183013_b(chunkpos, nbttagcompound);
@@ -171,17 +252,19 @@
                     {
                         field_151505_a.error("Failed to save chunk", (Throwable)exception);
                     }
-                }
+             //   }
 
-                lvt_3_1_ = true;
-            }
-            finally
-            {
-                this.field_193415_c.remove(chunkpos);
-            }
+            //    lvt_3_1_ = true;
+            //}
+            //finally
+            //{
+            //    this.chunksBeingSaved.remove(chunkpos);
+            //}
+            retireChunkToWrite(chunkpos, nbttagcompound);
 
-            return lvt_3_1_;
-        }
+            //return lvt_3_1_;
+            return true;
+        //} //CM fixed indentation
     }
 
     private void func_183013_b(ChunkPos p_183013_1_, NBTTagCompound p_183013_2_) throws IOException
@@ -249,7 +332,8 @@
         });
     }
 
-    private void func_75820_a(Chunk p_75820_1_, World p_75820_2_, NBTTagCompound p_75820_3_)
+    // CM: change access to public & static
+    public static void func_75820_a(Chunk p_75820_1_, World p_75820_2_, NBTTagCompound p_75820_3_)
     {
         p_75820_3_.func_74768_a("xPos", p_75820_1_.field_76635_g);
         p_75820_3_.func_74768_a("zPos", p_75820_1_.field_76647_h);
@@ -295,6 +379,10 @@
         }
 
         p_75820_3_.func_74782_a("Sections", nbttaglist);
+        // NewLight PHIPRO-CARPET
+        //if(CarpetSettings.newLight){
+        //	carpet.helpers.LightingHooks.writeLightData(chunkIn, compound);
+        //}
         p_75820_3_.func_74773_a("Biomes", p_75820_1_.func_76605_m());
         p_75820_1_.func_177409_g(false);
         NBTTagList nbttaglist1 = new NBTTagList();
@@ -347,19 +435,19 @@
         }
     }
 
-    private Chunk func_75823_a(World p_75823_1_, NBTTagCompound p_75823_2_)
+    private Chunk readChunkFromNBT(World worldIn, NBTTagCompound compound, boolean silent)
     {
-        int i = p_75823_2_.func_74762_e("xPos");
-        int j = p_75823_2_.func_74762_e("zPos");
-        Chunk chunk = new Chunk(p_75823_1_, i, j);
-        chunk.func_177420_a(p_75823_2_.func_74759_k("HeightMap"));
-        chunk.func_177446_d(p_75823_2_.func_74767_n("TerrainPopulated"));
-        chunk.func_177421_e(p_75823_2_.func_74767_n("LightPopulated"));
-        chunk.func_177415_c(p_75823_2_.func_74763_f("InhabitedTime"));
-        NBTTagList nbttaglist = p_75823_2_.func_150295_c("Sections", 10);
+        int i = compound.func_74762_e("xPos");
+        int j = compound.func_74762_e("zPos");
+        Chunk chunk = new Chunk(worldIn, i, j);
+        chunk.func_177420_a(compound.func_74759_k("HeightMap"));
+        chunk.func_177446_d(compound.func_74767_n("TerrainPopulated"));
+        chunk.func_177421_e(compound.func_74767_n("LightPopulated"));
+        chunk.func_177415_c(compound.func_74763_f("InhabitedTime"));
+        NBTTagList nbttaglist = compound.func_150295_c("Sections", 10);
         int k = 16;
         ExtendedBlockStorage[] aextendedblockstorage = new ExtendedBlockStorage[16];
-        boolean flag = p_75823_1_.field_73011_w.func_191066_m();
+        boolean flag = worldIn.field_73011_w.func_191066_m();
 
         for (int l = 0; l < nbttaglist.func_74745_c(); ++l)
         {
@@ -383,26 +471,39 @@
 
         chunk.func_76602_a(aextendedblockstorage);
 
-        if (p_75823_2_.func_150297_b("Biomes", 7))
+        // NewLight PHIPRO-CARPET
+        //if(CarpetSettings.newLight){
+        //	carpet.helpers.LightingHooks.readLightData(chunk, compound);
+        //}
+
+        if (compound.func_150297_b("Biomes", 7))
         {
-            chunk.func_76616_a(p_75823_2_.func_74770_j("Biomes"));
+            chunk.func_76616_a(compound.func_74770_j("Biomes"));
         }
 
-        NBTTagList nbttaglist1 = p_75823_2_.func_150295_c("Entities", 10);
+        // CM start
+        // carpet12RNY silent loading
+        if (silent)
+        {
+            return chunk;
+        }
+        // CM end
 
+        NBTTagList nbttaglist1 = compound.func_150295_c("Entities", 10);
+
         for (int j1 = 0; j1 < nbttaglist1.func_74745_c(); ++j1)
         {
             NBTTagCompound nbttagcompound1 = nbttaglist1.func_150305_b(j1);
-            func_186050_a(nbttagcompound1, p_75823_1_, chunk);
+            func_186050_a(nbttagcompound1, worldIn, chunk);
             chunk.func_177409_g(true);
         }
 
-        NBTTagList nbttaglist2 = p_75823_2_.func_150295_c("TileEntities", 10);
+        NBTTagList nbttaglist2 = compound.func_150295_c("TileEntities", 10);
 
         for (int k1 = 0; k1 < nbttaglist2.func_74745_c(); ++k1)
         {
             NBTTagCompound nbttagcompound2 = nbttaglist2.func_150305_b(k1);
-            TileEntity tileentity = TileEntity.func_190200_a(p_75823_1_, nbttagcompound2);
+            TileEntity tileentity = TileEntity.func_190200_a(worldIn, nbttagcompound2);
 
             if (tileentity != null)
             {
@@ -410,9 +511,9 @@
             }
         }
 
-        if (p_75823_2_.func_150297_b("TileTicks", 9))
+        if (compound.func_150297_b("TileTicks", 9))
         {
-            NBTTagList nbttaglist3 = p_75823_2_.func_150295_c("TileTicks", 10);
+            NBTTagList nbttaglist3 = compound.func_150295_c("TileTicks", 10);
 
             for (int l1 = 0; l1 < nbttaglist3.func_74745_c(); ++l1)
             {
@@ -428,7 +529,7 @@
                     block = Block.func_149729_e(nbttagcompound3.func_74762_e("i"));
                 }
 
-                p_75823_1_.func_180497_b(new BlockPos(nbttagcompound3.func_74762_e("x"), nbttagcompound3.func_74762_e("y"), nbttagcompound3.func_74762_e("z")), block, nbttagcompound3.func_74762_e("t"), nbttagcompound3.func_74762_e("p"));
+                worldIn.func_180497_b(new BlockPos(nbttagcompound3.func_74762_e("x"), nbttagcompound3.func_74762_e("y"), nbttagcompound3.func_74762_e("z")), block, nbttagcompound3.func_74762_e("t"), nbttagcompound3.func_74762_e("p"));
             }
         }
 
@@ -521,6 +622,10 @@
 
     public static void func_186052_a(Entity p_186052_0_, World p_186052_1_)
     {
+        // CM lifetime tracker
+        // this method is only used in MobSpawnerBaseLogic
+        p_186052_0_.recordSpawning(LiteralSpawningReason.SPAWNER);
+
         if (p_186052_1_.func_72838_d(p_186052_0_) && p_186052_0_.func_184207_aI())
         {
             for (Entity entity : p_186052_0_.func_184188_bt())
