--- ../src-base/minecraft/net/minecraft/village/VillageCollection.java
+++ ../src-work/minecraft/net/minecraft/village/VillageCollection.java
@@ -1,5 +1,7 @@
 package net.minecraft.village;
 
+import carpet.logging.LoggerRegistry;
+import carpet.logging.logHelpers.RNGMonitor;
 import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
@@ -14,8 +16,11 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldProvider;
+import net.minecraft.world.WorldServer;
 import net.minecraft.world.storage.WorldSavedData;
 
+import carpet.carpetclient.CarpetClientMarkers;
+
 public class VillageCollection extends WorldSavedData
 {
     private World field_75556_a;
@@ -23,6 +28,8 @@
     private final List<VillageDoorInfo> field_75555_c = Lists.<VillageDoorInfo>newArrayList();
     private final List<Village> field_75552_d = Lists.<Village>newArrayList();
     private int field_75553_e;
+    // Added update markers to update the client when a new door or village is added to the village list CARPET-XCOM
+    public static boolean updateMarkers;
 
     public VillageCollection(String p_i1677_1_)
     {
@@ -60,6 +67,12 @@
     public void func_75544_a()
     {
         ++this.field_75553_e;
+        updateMarkers = false;
+        // CM start
+        if (LoggerRegistry.__rngManip && field_75556_a instanceof WorldServer) {
+            ((WorldServer) field_75556_a).rngMonitor.tryUpdateSeeds(RNGMonitor.RNGAppType.ironFarm);
+        }
+        // CM end
 
         for (Village village : this.field_75552_d)
         {
@@ -69,7 +82,9 @@
         this.func_75549_c();
         this.func_75543_d();
         this.func_75545_e();
-
+        //Update villager markers on carpet client CARPET-XCOM
+        if(updateMarkers) CarpetClientMarkers.updateClientVillageMarkers(field_75556_a);
+        
         if (this.field_75553_e % 400 == 0)
         {
             this.func_76185_a();
@@ -88,6 +103,7 @@
             {
                 iterator.remove();
                 this.func_76185_a();
+                updateMarkers = true;
             }
         }
     }
@@ -144,6 +160,7 @@
             }
 
             village.func_75576_a(villagedoorinfo);
+            updateMarkers = true;
         }
 
         this.field_75555_c.clear();
@@ -277,6 +294,7 @@
             Village village = new Village();
             village.func_82690_a(nbttagcompound);
             this.field_75552_d.add(village);
+            updateMarkers = true;
         }
     }
 
