--- ../src-base/minecraft/net/minecraft/pathfinding/PathFinder.java
+++ ../src-work/minecraft/net/minecraft/pathfinding/PathFinder.java
@@ -1,5 +1,7 @@
 package net.minecraft.pathfinding;
 
+import carpet.logging.LoggerRegistry;
+import carpet.logging.logHelpers.PathReporter;
 import com.google.common.collect.Sets;
 import java.util.Set;
 import javax.annotation.Nullable;
@@ -40,6 +42,11 @@
         PathPoint pathpoint = this.field_176190_c.func_186318_b();
         PathPoint pathpoint1 = this.field_176190_c.func_186325_a(p_186334_3_, p_186334_5_, p_186334_7_);
         Path path = this.func_186335_a(pathpoint, pathpoint1, p_186334_9_);
+        // CM start
+        if (LoggerRegistry.__pathFinding) {
+            PathReporter.report(path);
+        }
+        // CM end
         this.field_176190_c.func_176163_a();
         return path;
     }
