--- ../src-base/minecraft/net/minecraft/block/state/BlockStateContainer.java
+++ ../src-work/minecraft/net/minecraft/block/state/BlockStateContainer.java
@@ -1,5 +1,6 @@
 package net.minecraft.block.state;
 
+import carpet.utils.SilentChunkReader;
 import com.google.common.base.Function;
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.HashBasedTable;
@@ -435,5 +436,11 @@
             {
                 return this.field_177239_a.func_193383_a(p_193401_1_, this, p_193401_2_, p_193401_3_);
             }
+
+            // carpet12RNY - SilentChunkReader
+            @Override
+            public void addCollisionBoxToList_silent(SilentChunkReader access, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn, boolean isActualState) {
+                field_177239_a.addCollisionBoxToList_silent(this, access, pos, entityBox, collidingBoxes, entityIn, isActualState);
+            }
         }
 }
