--- ../src-base/minecraft/net/minecraft/block/BlockLilyPad.java
+++ ../src-work/minecraft/net/minecraft/block/BlockLilyPad.java
@@ -68,4 +68,11 @@
     {
         return 0;
     }
+
+    // carpet12RNY - SilentChunkReader
+    public void addCollisionBoxToList_silent(IBlockState state, IBlockAccess access, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn, boolean isActualState) {
+        if (!(entityIn instanceof EntityBoat)) {
+            func_185492_a(pos, entityBox, collidingBoxes, field_185523_a);
+        }
+    }
 }
