--- ../src-base/minecraft/net/minecraft/block/BlockPistonMoving.java
+++ ../src-work/minecraft/net/minecraft/block/BlockPistonMoving.java
@@ -3,6 +3,8 @@
 import java.util.List;
 import java.util.Random;
 import javax.annotation.Nullable;
+
+import carpet.utils.SilentChunkReader;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyDirection;
@@ -214,4 +216,12 @@
     {
         return BlockFaceShape.UNDEFINED;
     }
+
+    // carpet12RNY - SilentChunkReader
+    public void addCollisionBoxToList_silent(IBlockState state, SilentChunkReader access, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn, boolean isActualState) {
+        //TileEntityPiston tileentitypiston = getTilePistonAt(access, pos);
+        //if (tileentitypiston != null) {
+        //    tileentitypiston.addCollissionAABBs(worldIn, pos, entityBox, collidingBoxes, entityIn);
+        //}
+    }
 }
