--- ../src-base/minecraft/net/minecraft/block/BlockBrewingStand.java
+++ ../src-work/minecraft/net/minecraft/block/BlockBrewingStand.java
@@ -178,4 +178,10 @@
     {
         return BlockFaceShape.UNDEFINED;
     }
+
+    // carpet12RNY - SilentChunkReader
+    public void addCollisionBoxToList_silent(IBlockState state, IBlockAccess access, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn, boolean isActualState) {
+        func_185492_a(pos, entityBox, collidingBoxes, field_185556_c);
+        func_185492_a(pos, entityBox, collidingBoxes, field_185555_b);
+    }
 }
